src/ExternalImage.php

```diff
@@ -0,0 +1,479 @@
+<?php
+
+namespace Timber;
+
+/**
+ * Class ExternalImage
+ *
+ * The `Timber\ExternalImage` class represents an external image loaded via http.
+ *
+ * @api
+ * @example
+ * ```php
+ * $context = Timber::context();
+ *
+ * // Lets say you have an external image that you want to use in your theme
+ *
+ * $context['cover_image'] = Timber::get_external_image($url);
+ *
+ * Timber::render('single.twig', $context);
+ * ```
+ *
+ * ```twig
+ * <article>
+ *   <img src="{{cover_image.src}}" class="cover-image" />
+ *   <h1 class="headline">{{post.title}}</h1>
+ *   <div class="body">
+ *     {{post.content}}
+ *   </div>
+ *
+ *  <img
+ *    src="{{ get_image(post.custom_field_with_image_id).src }}"
+ *    alt="Another way to initialize images as Timber\Image objects, but within Twig" />
+ * </article>
+ * ```
+ *
+ * ```html
+ * <article>
+ *   <img src="http://example.org/wp-content/uploads/2015/06/nevermind.jpg" class="cover-image" />
+ *   <h1 class="headline">Now you've done it!</h1>
+ *   <div class="body">
+ *     Whatever whatever
+ *   </div>
+ *   <img
+ *     src="http://example.org/wp-content/uploads/2015/06/kurt.jpg"
+ *     alt="Another way to initialize images as Timber\Image objects, but within Twig" />
+ * </article>
+ * ```
+ */
+class ExternalImage implements ImageInterface
+{
+    /**
+     * File.
+     *
+     * @api
+     * @var string
+     */
+    private $alt_text;
+
+    protected function __construct($alt)
```

Comment:
I think I wouldn’t use alt in the constructor here, but introduce a `set_alt()` method so that we could also set an alt text from outside.

```php
$image = Timber::get_external_image($url);
$image->set_alt('Dogs are happy outside');
```

I would probably even not use the constructor at all. The pattern that we use for the factories for WordPress objects in Timber is that we use `new static()`, see https://github.com/timber/timber/blob/1b91b3c31589162b2bc1872d0ff946fa7b68665d/src/Post.php#L181.

Reply:
Makes sense. Updated both.

However, I left `$alt` as an optional second parameter in `get_external_image`, as all images should really have an alt text (even if it's empty).